dashboard/app/views/devise/registrations/edit.html.haml

```diff
@@ -60,7 +60,7 @@
   -# Don't allow editing user type unless we can also edit email, because
   -# changing from a student (encrypted email) to a teacher (plaintext email)
   -# requires entering an email address.
-  - if f.object.can_edit_email?
+  - if f.object.can_edit_email? && Section.where(user_id: f.object.id).empty?
```

Comment:
Note: This will also prevent someone already in a bad state (a student that has sections) from changing their account to a teacher account.  Is that also desired behavior?

Also, please add a comment (like the one for the `can_edit_email?` restriction above) because I think the connection between the "user sections empty" check and showing the `user_type` field will not be obvious to future-me.